api/v1alpha1/nutanix_clusterconfig_types.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// NutanixSpec defines the desired state of NutanixCluster.
+type NutanixSpec struct {
+	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// host can be either DNS name or ip address
+	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// Nutanix Prism Central endpoint configuration.
+	PrismCentralEndpoint NutanixPrismCentralEndpointSpec `json:"prismCentralEndpoint"`
+}
+
+func (NutanixSpec) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Description: "Nutanix cluster configuration",
+			Type:        "object",
+			Properties: map[string]clusterv1.JSONSchemaProps{
+				"controlPlaneEndpoint": ControlPlaneEndpointSpec{}.VariableSchema().OpenAPIV3Schema,
+				"prismCentralEndpoint": NutanixPrismCentralEndpointSpec{}.VariableSchema().OpenAPIV3Schema,
+			},
+		},
+	}
+}
+
+type NutanixPrismCentralEndpointSpec struct {
+	// address is the endpoint address (DNS name or IP address) of the Nutanix Prism Central
+	Address string `json:"address"`
+
+	// port is the port number to access the Nutanix Prism Central
+	Port int32 `json:"port"`
+
+	// use insecure connection to Prism Central endpoint
+	// +optional
+	Insecure bool `json:"insecure"`
+
+	// A reference to the ConfigMap containing a PEM encoded x509 cert for the RootCA that was used to create
+	// the certificate for a Prism Central that uses certificates that were issued by a non-publicly trusted RootCA.
+	// The trust bundle is added to the cert pool used to authenticate the TLS connection to the Prism Central.
+	// +optional
+	AdditionalTrustBundle *corev1.LocalObjectReference `json:"additionalTrustBundle,omitempty"`
+
+	// A reference to the Secret for credential information for the target Prism Central instance
+	Credentials corev1.LocalObjectReference `json:"credentials"`
```

Comment:
why not secretRef?

Reply:
Consistency with existing API https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/blob/47e578a14dc9855ab8f76dbfa087c20972fd41db/api/v1alpha1/clusterconfig_types.go#L253 

I'm also not sure we do want cross namespace refs here, it would complicate things when needing to move these Secrets to the management cluster. It also removes the ambiguity of what an empty namespace means, is it the default namespace or the Cluster's namespace?

Reply:
In this case this secret would need to exist in the default namespace, I don't think we want that